acme/api/handler.go

```diff
@@ -184,10 +215,25 @@ func (h *Handler) GetChallenge(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	getLink := h.Auth.GetLink
-	w.Header().Add("Link", link(getLink(acme.AuthzLink, acme.URLSafeProvisionerName(prov), true, ch.GetAuthzID()), "up"))
-	w.Header().Set("Location", getLink(acme.ChallengeLink, acme.URLSafeProvisionerName(prov), true, ch.GetID()))
-	api.JSON(w, ch)
+	switch ch.Status {
+	case acme.StatusPending:
+		panic("validation attempt did not move challenge to the processing state")
+	// When a transient error occurs, the challenge will not be progressed to the `invalid` state.
+	// Add a Retry-After header to indicate that the client should check again in the future.
+	case acme.StatusProcessing:
+		w.Header().Add("Retry-After", ch.RetryAfter)
+		w.Header().Add("Cache-Control", "no-cache")
+		api.JSON(w, ch)
+	case acme.StatusInvalid:
+		api.JSON(w, ch)
+	case acme.StatusValid:
+		getLink := h.Auth.GetLink
+		w.Header().Add("Link", link(getLink(acme.AuthzLink, acme.URLSafeProvisionerName(prov), true, ch.GetAuthzID()), "up"))
```

Comment:
Why are we only setting `link` and `location` for valid challenges?

Reply:
I think previously we would have written out the acme error. I'm not familiar with the `Link` and `Location` headers WRT ACME. Should then be returned on all valid resource gets?

Reply:
Yeah, I think we should return these with any 2xx response.